calm-integrations/generate_task_library_items/generate_task_library.py

```diff
@@ -1,79 +1,153 @@
+    parser.add_argument('-p', '--project',
+                        required=True,
+                        action='store',
+                        help='Project to which task library item is created ')
+    parser.add_argument('-s', '--script',
+                        required=True,
+                        action='store',
+                        help='Script path')
+    return parser
+# --------------------------------------------------------------------------------- #
 
-script_path = sys.argv[1]
-project_name = "default"
-project_uuid = "7ffa7e0c-0929-41b7-bb30-4dfe836e8435"
-
-script_content = ""
-
-try:
-    scriptf = open(script_path, "r")
-except IOError:
-    print("ERROR: File not found.")
-    sys.exit(1)
-finally:
-    script_content = scriptf.read()
-
-script_name = ntpath.basename(script_path)
-item_name = os.path.splitext(script_name.replace("_", ' '))[0]
-script_type = os.path.splitext(script_name)[1].replace('.', '')
-
-payload = {
-  "api_version": "3.0",
-  "metadata": {
-    "kind": "app_task",
-    "project_reference": {
-      "kind": "project",
-      "name": project_name,
-      "uuid": project_uuid
+
+def get_project_uuid(base_url, auth, project_name):
+    method = 'POST'
+    url = base_url + "/projects/list"
+    payload = {
+        "length": 100,
+        "offset": 0,
+        "filter": "name=={0}".format(project_name)
     }
-  },
-  "spec": {
-    "name": item_name,
-    "resources": {
-      "variable_list": [],
-      "attrs": {
-        "script": script_content,
-        "script_type": script_type
-      },
-      "type": "EXEC"
-    },
-    "description": ""
-  }
-}
+    resp = requests.request(
+        method,
+        url,
+        data=json.dumps(payload),
+        headers=headers,
+        auth=(auth["username"], auth["password"]),
+        verify=False
+    )
 
-pc_ip = "10.0.0.0"
-pc_username = "admin"
-pc_password = "secret"
+    if resp.ok:
+        json_resp = resp.json()
+        if json_resp['metadata']['total_matches'] > 0:
+            project = json_resp['entities'][0]
+            project_uuid = project["metadata"]["uuid"]
+            return project_uuid
+        else:
+            logging.error("Could not find project")
+            sys.exit(-1)
+    else:
+        logging.error("Request failed")
+        logging.error("Headers: {}".format(headers))
+        logging.error('Status code: {}'.format(resp.status_code))
+        logging.error('Response: {}'.format(json.dumps(json.loads(resp.content), indent=4)))
+        sys.exit(-1)
 
-if pc_ip and pc_username and pc_password:
-    import requests
+# --------------------------------------------------------------------------------- #
 
-    url = "https://{}:9440/api/nutanix/v3/app_tasks".format(pc_ip)
+
+def seed_task_item(base_url, auth, project_name, path):
+    print("INFO: Path of script '{}'.".format(path))
+    script_content = ""
+    try:
+        scriptf = open(script_path, "r")
```

Comment:
Did not test this part but speculating `script_path` would be unknown to `seed_task_item` function.

Reply:
script path is one of the argument to seed_task_item